src/transformers/models/gptj/modeling_gptj.py

```diff
@@ -0,0 +1,845 @@
+                q_pass = query[:, :, :, self.rotary_dim :]
+
+                sincos = fixed_pos_embedding(k_rot, 1, seq_len=seq_len)
+                k_rot = apply_rotary_pos_emb(k_rot, sincos, offset=offset)
+                q_rot = apply_rotary_pos_emb(q_rot, sincos, offset=offset)
+
+                key = torch.cat([k_rot, k_pass], dim=-1)
+                query = torch.cat([q_rot, q_pass], dim=-1)
+            else:
+                sincos = fixed_pos_embedding(key, 1, seq_len=seq_len)
+                key = apply_rotary_pos_emb(key, sincos, offset=offset)
+                query = apply_rotary_pos_emb(query, sincos, offset=offset)
+            key = key.permute(0, 2, 1, 3)
+            query = query.permute(0, 2, 1, 3)
+
+        if layer_past is not None:
+            past_key = layer_past[0]
+            past_value = layer_past[1]
+            key = torch.cat((past_key, key), dim=-2)
+            value = torch.cat((past_value, value), dim=-2)
+
+        if use_cache is True:
+            present = (key, value)
+        else:
+            present = None
+
+        query_length, key_length = query.size(-2), key.size(-2)
+        causal_mask = self.bias[:, :, key_length - query_length : key_length, :key_length]
+
+        attn_output, attn_weights = self._attn(
+            query,
+            key,
+            value,
+            causal_mask,
+            self.masked_bias,
+            self.attn_dropout,
+            attention_mask,
+            head_mask,
+            self.scale_attn,
+        )
+
+        attn_output = self._merge_heads(attn_output, self.num_attention_heads, self.head_dim)
+        attn_output = self.out_proj(attn_output)
+        attn_output = self.resid_dropout(attn_output)
+
+        outputs = (attn_output, present)
+        if output_attentions:
+            outputs += (attn_weights,)
+
+        return outputs  # a, present, (attentions)
+
+
+class GPTJMLP(nn.Module):
+    def __init__(self, intermediate_size, config):  # in MLP: intermediate_size= 4 * embed_dim
+        super().__init__()
+        embed_dim = config.n_embd
+        self.c_fc = nn.Linear(embed_dim, intermediate_size)
+        self.c_proj = nn.Linear(intermediate_size, embed_dim)
+        self.act = ACT2FN[config.activation_function]
+        self.dropout = nn.Dropout(config.resid_pdrop)
+
+    def forward(self, hidden_states):
+        hidden_states = self.c_fc(hidden_states)
+        hidden_states = self.act(hidden_states)
+        hidden_states = self.c_proj(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        return hidden_states
+
+
+class GPTJBlock(nn.Module):
+    def __init__(self, config, layer_id):
+        super().__init__()
+        inner_dim = config.intermediate_size if config.intermediate_size is not None else 4 * config.n_embd
+        self.ln_1 = nn.LayerNorm(config.n_embd, eps=config.layer_norm_epsilon)
+        self.attn = GPTJAttention(config, layer_id)
```

Comment:
@StellaAthena I think here we should delete `layer_id` as it's not expected anymore by the class `GPTJAttention` :-)